.github/workflows/ci.yml

```diff
@@ -19,6 +20,18 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  preconditions:
+    name: Preconditions
+    runs-on: ubuntu-latest
+    outputs:
+      notebooks: ${{ steps.filter.outputs.notebooks }}
+      e2e_tests: ${{ steps.filter.outputs.e2e_tests }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
```

Comment:
what is this ?
dorny ? is it a private user ?
we shouldn't depend on unknown stuff,
maybe we can copy/paste the action ?

Reply:
well, it's a pretty established action: https://github.com/dorny/paths-filter

I don't see any reason for not using it. Also we're not passing to it any secret.